src/argus_htmx/destinations/views.py

```diff
@@ -1,81 +1,128 @@
+        return destinations_list(request)
+    elif request.method == "POST":
+        return destinations_create(request)
+
+
+@require_GET
+def destinations_list(request) -> HttpResponse:
+    forms = _get_destination_forms_grouped_by_media(request.user)
     context = {
-        "form": DestinationForm(),
-        "grouped_destinations": destinations,
+        "form": DestinationFormCreate(),
+        "grouped_forms": forms,
     }
     return render(request, "htmx/destinations/destinations.html", context=context)
 
 
-def _get_destinations_and_forms_grouped_by_media(user) -> dict[Media, list[DestinationConfig]]:
+def _get_destination_forms_grouped_by_media(user) -> dict[Media, list[DestinationFormUpdate]]:
     """Returns dict where key is media and value is list of tuples
     containing a destination and a pre-filled form for that destination."""
     grouped_destinations = {}
-
     media = Media.objects.all()
     for m in media:
         grouped_destinations[m] = []
 
     destinations = user.destinations.all()
     for destination in destinations:
-        form = DestinationForm(
+        settings_key = _get_settings_key_for_media(destination.media)
+        form = DestinationFormUpdate(
+            instance=destination,
             initial={
-                "label": destination.label,
-                "media": destination.media.slug,
-                "value": destination.settings.get("email_address", ""),
-            }
+                "value": destination.settings.get(settings_key, ""),
+            },
         )
-        form.fields["media"].widget.attrs["hidden"] = True
-        grouped_destinations[destination.media].append((destination, form))
+        grouped_destinations[destination.media].append(form)
 
     return grouped_destinations
 
 
 @require_POST
 def destinations_create(request) -> HttpResponse:
-    form = DestinationForm(request.POST or None)
+    form = DestinationFormCreate(request.POST or None)
     if form.is_valid():
         media = Media.objects.get(slug=form.cleaned_data["media"])
-        medium = api_safely_get_medium_object(media.slug)
-        # e.g. "email_address", "phone_number" etc.
-        settings_key = medium.MEDIA_JSON_SCHEMA["required"][0]
-        destination = DestinationConfig.objects.create(
-            user=request.user,
-            media=media,
-            label=form.cleaned_data.get("label", ""),
-            settings={settings_key: form.cleaned_data["value"]},
+        settings_key = _get_settings_key_for_media(media)
+
+        serializer = RequestDestinationConfigSerializer(
+            data={
+                "media": media,
+                "label": form.cleaned_data.get("label", ""),
+                "settings": {settings_key: form.cleaned_data["value"]},
+            },
+            context={"request": request},
         )
-        destination.save()
+        serializer.is_valid(raise_exception=True)
+        serializer.save(user=request.user)
+
```

Comment:
I was thinking you'd use the form to save, not go via a serializer. There's magic needed but we all ought to know that magic.